cmd/basecli/commands/cmds.go

```diff
@@ -0,0 +1,200 @@
+package commands
+
+import (
+	"encoding/hex"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/tendermint/light-client/commands"
+	txcmd "github.com/tendermint/light-client/commands/txs"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	cmn "github.com/tendermint/tmlibs/common"
+
+	btypes "github.com/tendermint/basecoin/types"
+)
+
+/******** SendTx *********/
+
+// SendTxCmd is CLI command to send tokens between basecoin accounts
+var SendTxCmd = &cobra.Command{
+	Use:   "send",
+	Short: "send tokens from one account to another",
+	RunE:  doSendTx,
+}
+
+//nolint
+const (
+	FlagTo       = "to"
+	FlagAmount   = "amount"
+	FlagFee      = "fee"
+	FlagGas      = "gas"
+	FlagSequence = "sequence"
+)
+
+func init() {
+	flags := SendTxCmd.Flags()
+	flags.String(FlagTo, "", "Destination address for the bits")
+	flags.String(FlagAmount, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
+	flags.String(FlagFee, "0mycoin", "Coins for the transaction fee of the format <amt><coin>")
+	flags.Int64(FlagGas, 0, "Amount of gas for this transaction")
+	flags.Int(FlagSequence, -1, "Sequence number for this transaction")
+}
+
+// runDemo is an example of how to make a tx
+func doSendTx(cmd *cobra.Command, args []string) error {
+
+	// load data from json or flags
+	tx := new(btypes.SendTx)
+	found, err := txcmd.LoadJSON(tx)
+	if err != nil {
+		return err
+	}
+	if !found {
+		err = readSendTxFlags(tx)
+	}
+	if err != nil {
+		return err
+	}
+
+	// Wrap and add signer
+	send := &SendTx{
+		chainID: commands.GetChainID(),
+		Tx:      tx,
+	}
+	send.AddSigner(txcmd.GetSigner())
+
+	// Sign if needed and post.  This it the work-horse
+	bres, err := txcmd.SignAndPostTx(send)
+	if err != nil {
+		return err
+	}
+
+	// Output result
+	return txcmd.OutputTx(bres)
+}
+
+func readSendTxFlags(tx *btypes.SendTx) error {
+	// parse to address
+	to, err := parseChainAddress(viper.GetString(FlagTo))
+	if err != nil {
+		return err
+	}
+
+	//parse the fee and amounts into coin types
+	tx.Fee, err = btypes.ParseCoin(viper.GetString(FlagFee))
+	if err != nil {
+		return err
+	}
+	amountCoins, err := btypes.ParseCoins(viper.GetString(FlagAmount))
+	if err != nil {
+		return err
+	}
+
+	// set the gas
+	tx.Gas = viper.GetInt64(FlagGas)
+
+	// craft the inputs and outputs
+	tx.Inputs = []btypes.TxInput{{
+		Coins:    amountCoins,
+		Sequence: viper.GetInt(FlagSequence),
```

Comment:
Noted: https://github.com/tendermint/basecoin/issues/104

Open discussion on how best to do this.  Querying every time seems a bit ugly for me, but can be done.  Please add discussion there and let's target next release.